frontend/components/charities/CharityDeletionDialog.tsx

```diff
@@ -0,0 +1,36 @@
+import { Dialog, DialogActions, DialogContent, DialogTitle, Typography } from '@mui/material';
+import Button from '../generic/Button';
+import { CharityListData } from '../../types/charity';
+
+export default function CharityDeletionDialog({
+  selectedCharity,
+  open,
+  handleClose,
+  handleConfirm,
+}: {
+  selectedCharity: CharityListData;
+  open: boolean;
+  handleClose: () => void;
+  handleConfirm: (selectedCharity: CharityListData) => void;
+}) {
+  const { name } = selectedCharity;
+
+  return (
+    <Dialog open={open} onClose={handleClose}>
+      <DialogTitle>
+        <Typography variant="h1">Delete {name}</Typography>
+      </DialogTitle>
+
+      <DialogContent>Are you sure you want to delete this charity? This cannot be undone.</DialogContent>
+
+      <DialogActions>
+        <Button actionType="muted" onClick={handleClose}>
+          Cancel
+        </Button>
+        <Button actionType="danger" onClick={() => handleConfirm(selectedCharity)}>
```

Comment:
The parent would have the charityId,
A `handleDelete: () => void` allows this delete dialog to be more reusable for other cases

```suggestion
        <Button actionType="danger" onClick={handleDelete}>
```